### PR TITLE
Misc small fixes

### PR DESCRIPTION
**Pull request by steve-downey**
HIghlight dead code in in the c-ts-mode
Clangd binary names update
Org mode from elpa
Elpa compilation dir not hardcoded to ~/.emacs.d/

### CODE REVIEW

File: init.el [hunk 1]
```diff
@@ -209,6 +210,12 @@ Each element of the list is in the same form as in `package-pinned-packages'."
 (require 'use-package-ensure)
 (setq use-package-always-ensure t)
 (setq use-package-compute-statistics t)
+
+;;; remove a package from the builtin list so it can be upgraded
+(defun exordium-ignore-builtin (pkg)
+  (assq-delete-all pkg package--builtins)
+  (assq-delete-all pkg package--builtin-versions))
+
```
Comment (pkryger) [inline init.el:218]: I've branched out a bit, but using a built-in functionality, like [here](https://github.com/pkryger/exordium/commit/4560435403e1129c4b36c30c19e5f37906a2bebc). Any reasons for not allowing all packages to be upgradeable?

File: modules/init-cpp.el [hunk 2]
```diff
@@ -49,7 +49,8 @@
 
 ;;; Highlight dead code between "#if 0" and "#endif"
 (add-hook 'c-mode-common-hook 'cpp-highlight-dead-code-hook)
-
+(when exordium-treesit-modes-enable
+  (add-hook 'c-ts--mode-common-hook 'cpp-highlight-dead-code-hook))
```
Comment (pkryger) [inline modules/init-cpp.el:53]: I wonder if using a forwarding hook, just like other hooks are done wouldn't be more appropriate, i.e., something like:
```
(add-hook 'c-ts--mode-common-hook #'(lambda () (run-hooks 'c-mode-common-hook))
```
But added in the [`init-exordium.el`](https://github.com/emacs-exordium/exordium/blob/20a4a819873678edc6332c8f193fb22941e18c58/modules/init-treesit.el#L51). That way the same approach will be used everywhere.

File: init.el (hunk 1 above)
Comment (steve-downey) [reply init.el:218]: I'd somehow come to the conclusion that this didn't work for the initial install of everything? Let me try an empty emacs.d and check which `org` I end up with. 
I don't see a strong reason not to allow all packages to be updated, but `org` was a particular pain point.

File: modules/init-cpp.el (hunk 2 above)
Comment (steve-downey) [reply modules/init-cpp.el:53]: Sounds reasonable. I'll try it out. I don't think I've found any hooks I don't want applied to ts and plain modes.

File: init.el (hunk 1 above)
Comment (steve-downey) [reply init.el:218]: Just rechecked. Removing it from the package-builtins list gets org-9.7pre0.20240513.124002 installed the first time, while (setq package-install-upgrade-built-in t) doesn't, at least initially. 
I have a few command-line workflows where I want the system to be able to initialize itself correctly in a different emacs.d directory.

File: modules/init-cpp.el (hunk 2 above)
Comment (steve-downey) [reply modules/init-cpp.el:53]: Looks like the ts modes don't wire up a common hook here, and need to attach the common hook to each mode.

**Review by pkryger — APPROVED** [review]
LGTM

**Pull request merged**